sqlmesh/core/plan/builder.py

```diff
@@ -554,22 +565,24 @@ def _check_destructive_changes(self, directly_modified: t.Set[SnapshotId]) -> No
             if columns_to_types_all_known(old_columns_to_types) and columns_to_types_all_known(
                 new_columns_to_types
             ):
-                schema_diff = get_schema_differ(snapshot.model.dialect).compare_columns(
-                    new.name,
-                    old_columns_to_types,
-                    new_columns_to_types,
-                    ignore_destructive=new.model.on_destructive_change.is_ignore,
+                alter_operations = t.cast(
+                    t.List[TableAlterOperation],
```

Comment:
I'm curious why do you need this case? Isn't return type of `compare_columns` already a subtype?